packages/consensus/src/validation/coinbase.cairo

```diff
@@ -105,14 +115,76 @@ fn compute_block_reward(block_height: u32) -> u64 {
     shr(5000000000_u64, (block_height / 210000_u32))
 }
 
+/// Calculate wtxid commitment
+fn calculate_wtxid_commitment(wtxid_root: Digest) -> Digest {
+    // construct witness reserved value
+    // 0000000000000000000000000000000000000000000000000000000000000000
+    let witness_value_byte: ByteArray =
+        "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0";
+
+    // convert wtxid_root to ByteArray
+    let wtxid_root_bytes: ByteArray = wtxid_root.into();
+
+    // concat (witness root hash | witness reserved value)
+    let mut res = ByteArrayTrait::concat(@wtxid_root_bytes, @witness_value_byte);
+
+    double_sha256_byte_array(@res)
+}
+
+/// validate segwit output (BIP-141)
+fn validate_segwit_output(
+    mut outputs: Span<TxOut>, wtxid_commitment: Digest
+) -> Result<(), ByteArray> {
+    let mut is_wtxid_commitment_present: bool = false;
+
+    // construct expected wtxid commitment
+    let mut fixed_prefix_byte: ByteArray = "";
+    fixed_prefix_byte.append_word(WTNS_PK_SCRIPT_PREFIX, 6);
+
+    let wtxid_commitment_bytes: ByteArray = wtxid_commitment.into();
+
+    let mut expected_wtxid_commitment = ByteArrayTrait::concat(
```

Comment:
You can probably just use one bytearray and append prefix and then the wtxid commitment ? That would be much more concise